src/views/transaction/components/transaction-form/transaction-form.view.jsx

```diff
@@ -52,6 +56,21 @@ function TransactionForm({
   const [doesReceiverExist, setDoesReceiverExist] = React.useState(undefined);
   const [doesUserHaveEnoughEthForWithdraw, setDoesUserHaveEnoughEthForWithdraw] =
     React.useState(undefined);
+  const [gasPrice, setGasPrice] = React.useState(BigNumber.from(0));
+  const [depositFee, setDepositFee] = React.useState(undefined);
+
+  React.useEffect(() => {
+    if (transactionType === TxType.Deposit) {
+      onLoadEstimatedDepositFee();
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (estimatedDepositFeeTask.status === "successful") {
+      setGasPrice(estimatedDepositFeeTask.data.gasPrice);
```

Comment:
Why do you need to set it instead of just using `estimatedDepositFeeTask` directly?

Reply:
To check status just one time and not introduce logic inside `AmountInput`